Added encode options

Superdavo0001 opened the pull request:
Added more resolutions, resolution-independent zoom (both from my mod), and fixed the "color tiggers" typo

Superdavo0001 committed:
Added encode options

Added more resolutions, resolution-independent zoom, and fixed the "color tiggers" typo

RatherBeLunar reviewed (changes requested):
@Superdavo0001, I don't think this is working. I selected the 4K option and got a 720p video. There is a boolean setting in Settings.cs called "Record1080p". This should be entirely removed as part of your change since the recording quality is no longer a toggle between two options, but I think it is also the reason your changes aren't working

Superdavo0001 committed:
Fixed 4k Encode Bug

...And replaced the switch statement & added more checks so it cannot happen again

RatherBeLunar committed:
Further optimize the resolution options with a dictionary
